pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -286,34 +283,22 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
 		memberStatusMap[c.Namespace] = c
 	}
 
-	// 3. create ServiceMeshMember object for each ns in spec.members
-	memberNamespaces := roll.Spec.Members
-	if roll.Spec.IsClusterScoped() {
-		nsList := &corev1.NamespaceList{}
-		err = r.Client.List(ctx, nsList)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		memberNamespaces = []string{}
-		for _, ns := range nsList.Items {
-			if ns.Name != meshNamespace && !isExcludedNamespace(ns.Name) {
-				memberNamespaces = append(memberNamespaces, ns.Name)
+	// 2. create ServiceMeshMember object for each ns in spec.members
+	if mesh != nil {
+		for _, ns := range requiredNamespaces.List() {
+			member, err := r.ensureMemberExists(ctx, ns, mesh.Name, meshNamespace)
+			if err != nil {
+				return reconcile.Result{}, err
```

Comment:
```suggestion
                                return common.RequeueWithError(err)
```

Reply:
I don't see any real benefit of doing this. I see how you could argue that seeing `RequeueWithError()` immediately tells you that the object is getting re-queued, but on the other hand, when you see this function call, you can't know what it does until you look at its implementation. 

I feel invoking a function here just slows down anyone reading this code (until they know by heart what the function does), whereas a simple `return reconcile.Result{}, err` perfectly shows what it does (even though it doesn't say that the object gets re-queued, but that's something every engineer working on k8s controllers should know). Also, who guarantees that the fact that you're returning an error will result in the object being re-queued? That may not be true in some cases. In this case, the function name would be misleading.

Reply:
> I don't see any real benefit of doing this.

Ok, I expected that you will like the idea, because I remember your positive feedback when I added this function. Look at this [comment](https://github.com/maistra/istio-operator/pull/910#discussion_r866839422).

> whereas a simple `return reconcile.Result{}, err` perfectly shows what it does (even though it doesn't say that the object gets re-queued, but that's something every engineer working on k8s controllers should know)

It's a subjective point of view and I don't agree with you. I review PRs regularly and every time I see reconciliation I have to stay focus to understand how we reconcile objects. And it's not only my opinion - look at point 7 [here](https://cloud.redhat.com/blog/7-best-practices-for-writing-kubernetes-operators-an-sre-perspective).

Reply:
I don't dislike the idea. I just think there are pros and cons (as always) that I outlined in my previous comment. And I agree with point 7, but also think it doesn't address the part about having to go look at the function (and possibly being misleading if the error is handled in one of the calling functions and thus preventing the re-queue).

The thing with these types of helper functions is that nothing forces you to use it, so you'll eventually end up with X% of the code using the function and the remaining code that doesn't. Then the question of consistency comes up. 

I don't really have a strong opinion on using one approach or the other. I guess I have to give your opinion a bit more weight, since I've been writing controllers a long time and it's all the same to me. But if it helps people who work on controller code less often, I agree that we should be using the function.

However, if we decide that we should always use `RequeueWithError` and similar, we should probably add a linter that enforces this decision.